boot/boot.go

```diff
@@ -0,0 +1,171 @@
+	ChangeRequiresReboot() bool
+}
+
+// Lookup figures out what the right boot set is for the given
+// arguments, and returns it.
+func Lookup(s snap.PlaceInfo, t snap.Type, onClassic bool) (DeviceBootSet, error) {
+	if t != snap.TypeOS && t != snap.TypeKernel && t != snap.TypeBase && t != snap.TypeSnapd {
+		return nil, fmt.Errorf("cannot lookup boot set with snap %q of type %q", s.SnapName(), t)
+	}
+
+	if onClassic {
+		return classicBootSet{}, nil
+	}
+
+	return &coreBootSet{s: s, t: t}, nil
+}
+
+type classicBootSet struct{}
+
+func (classicBootSet) RemoveKernelAssets() error {
+	return fmt.Errorf("cannot remove kernel assets on classic systems")
+}
+
+func (classicBootSet) ExtractKernelAssets(snap.Container) error {
+	return fmt.Errorf("cannot extract kernel assets on classic systems")
+}
+
+func (classicBootSet) SetNextBoot() error {
+	return fmt.Errorf("cannot set next boot on classic systems")
+}
+
+func (classicBootSet) ChangeRequiresReboot() bool { return false }
+
+var _ DeviceBootSet = classicBootSet{}
+
+// InUse checks if the given name/revision is used in the
+// boot environment
+func InUse(name string, rev snap.Revision) bool {
+	bootloader, err := bootloader.Find()
+	if err != nil {
+		logger.Noticef("cannot get boot settings: %s", err)
+		return false
+	}
+
+	bootVars, err := bootloader.GetBootVars("snap_kernel", "snap_try_kernel", "snap_core", "snap_try_core")
+	if err != nil {
+		logger.Noticef("cannot get boot vars: %s", err)
+		return false
+	}
+
+	snapFile := filepath.Base(snap.MountFile(name, rev))
+	for _, bootVar := range bootVars {
+		if bootVar == snapFile {
+			return true
+		}
+	}
+
+	return false
+}
+
+var (
+	ErrBootNameAndRevisionAgain = errors.New("boot revision not yet established")
+)
+
+type NameAndRevision struct {
+	Name     string
+	Revision snap.Revision
+}
+
+// GetCurrentBoot returns the currently set name and revision for boot for the given
+// type of snap, which can be snap.TypeBase (or snap.TypeOS), or snap.TypeKernel.
+// Returns ErrBootNameAndRevisionAgain if the values are temporarily not established.
+func GetCurrentBoot(t snap.Type) (*NameAndRevision, error) {
+	var bootVar, errName string
+	switch t {
+	case snap.TypeKernel:
+		bootVar = "snap_kernel"
+		errName = "kernel"
+	case snap.TypeOS, snap.TypeBase:
+		bootVar = "snap_core"
+		errName = "snap"
+	default:
+		return nil, fmt.Errorf("internal error: cannot find boot revision for snap type %q", t)
+	}
+
+	loader, err := bootloader.Find()
+	if err != nil {
+		return nil, fmt.Errorf("cannot get boot settings: %s", err)
+	}
+
+	m, err := loader.GetBootVars(bootVar, "snap_mode")
+	if err != nil {
+		return nil, fmt.Errorf("cannot get boot variables: %s", err)
+	}
+
+	if m["snap_mode"] == "trying" {
+		return nil, ErrBootNameAndRevisionAgain
+	}
+
+	nameAndRevno, err := nameAndRevnoFromSnap(m[bootVar])
+	if err != nil {
+		return nil, fmt.Errorf("cannot get name and revision of boot %s: %v", errName, err)
```

Comment:
This won't be i18n friendly as you cannot apply declination to the value of `errName`. I would suggest rewording this to avoid the problem.

Reply:
there is no i18n here.
Also, not new code.